src/rollback_to_stable/rts_btree.c

```diff
@@ -82,12 +82,28 @@ __rts_btree_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *firs
         }
     }
 
-    /*
-     * Clear the history store flags for the stable update to indicate that this update should be
-     * written to the history store later. The next time when this update is moved into the history
-     * store, it will have a different stop time point.
-     */
     if (stable_upd != NULL) {
+        /*
+         * During recovery, there shouldn't be any updates in the update chain except when the
+         * updates are from a prepared transaction. Reset the transaction ID of the stable update
+         * that was restored as part of the unstable prepared tombstone. Ignore the history store as
+         * we cannot have a prepared transaction operating on it.
+         */
+        if (!WT_IS_HS(session->dhandle) && F_ISSET(S2C(session), WT_CONN_RECOVERING)) {
+            WT_ASSERT(session, first_upd->type == WT_UPDATE_TOMBSTONE);
+            WT_ASSERT(session, first_upd->prepare_state == WT_PREPARE_INPROGRESS);
+            WT_ASSERT(session, F_ISSET(first_upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS));
+            WT_ASSERT(session, !hs_update);
```

Comment:
@quchenhao, we dont have access to the variable as it comes from on of your recent tickets that was not backported.
Should we remove the assertion or replace it with:
```suggestion
                        WT_ASSERT(session, !F_ISSET(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS));
```
I am not entirely sure as this flag is set while we traverse the update chain and each `upd` is checked, therefore, checking `stable_upd` may be wrong, i.e:
The first update has those flags but is not stable. The next update is stable but does not have this flag. Then the new assertion would fail.

Reply:
Yes, we can or we can just backport my previous change.

Reply:
I prefer to remove the assert instead of backporting many changes right now. I am fine to comment that assert and enable it later once we backport that change into 8.0 later.

Reply:
Sure we can do that.
The backport of WT-13198 is not requested though, is it the plan to backport it?